.github/workflows/osl_sync_main.yml

```diff
@@ -47,64 +43,54 @@ jobs:
+          pnpm bootstrap --no-frozen-lockfile
+          git add .
+          git merge --continue --no-edit
 
-      - name: Restore and Recreate Ignored Files
+      - name: Run Bootstrap (No Conflicts)
+        if: "!contains(steps.merge.outputs.result, 'CONFLICT')"
         run: |
-          for item in $IGNORED_FILES; do
-             [ -e "temp_ignored_items/$(basename "$item")" ] && mv -v "temp_ignored_items/$(basename "$item")" "$item"
-          done
-
-      - name: Setup environment
-        uses: ./.github/actions/setup-env
-
-      - name: Bootstrap (Regenerate Dependencies)
-        run: pnpm bootstrap --no-frozen-lockfile
+          pnpm bootstrap --no-frozen-lockfile
 
-      - name: "FULL → Build"
-        shell: bash
-        env:
-          KIE_TOOLS_BUILD__runTests: "false"
-          KIE_TOOLS_BUILD__runEndToEndTests: "false"
-          KIE_TOOLS_BUILD__buildContainerImages: "true"
-          KIE_TOOLS_BUILD__ignoreTestFailures: "true"
-          KIE_TOOLS_BUILD__ignoreEndToEndTestFailures: "true"
-          NODE_OPTIONS: "--max_old_space_size=4096"
+      - name: Build (Rebuild Files)
         run: >-
           eval "pnpm ${{ env.PNPM_FILTER }} --workspace-concurrency=1 build:prod"
 
       - name: Commit and Push Changes to Sync Branch
+        if: contains(steps.merge.outputs.result, 'CONFLICT') || git status --porcelain | grep -q '^[ACDM]'
         run: |
+          echo "Changes detected. Committing and pushing..."
+          git status --porcelain
           git add .
-          git commit -m "Sync with upstream/main and recreated ignored files"
+          git commit -m "Sync with upstream/main and resolved conflicts"
+          git push -u origin "${{ env.SYNC_BRANCH }}"
+
+      - name: Push Changes (No Conflict or Changes)
+        if: "!contains(steps.merge.outputs.result, 'CONFLICT') && !git status --porcelain | grep -q '^[ACDM]'"
+        run: |
+          echo "No conflicts or changes detected. Pushing merged branch."
           git push -u origin "${{ env.SYNC_BRANCH }}"
 
       - name: Create Pull Request
-        shell: bash
         env:
           RUN_URL: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
         run: |
-          set -x
-          # Use double quotes for variable substitution in sed
-          sed "s|\\\$RUN_URL|${RUN_URL}|g" .github/supporting-files/ci/templates/osl_sync_pr_template.md > temp.md
-          prTitle="[$(date +'%Y%m%d-%H%M%S')] - Automatic PR: Sync main with main-apache"
+          prTitle="[$(date +'%Y%m%d-%H%M%S')] - Automatic PR: Sync main with upstream"
+          reviewersOption=""
           if [[ -n "${{ vars.SYNC_REVIEWERS }}" ]]; then
             reviewersOption="--reviewer ${{ vars.SYNC_REVIEWERS }}"
           fi
-          gh pr create --title "${prTitle}" --body-file temp.md --repo kubesmarts/kie-tools --base main $reviewersOption
+          gh pr create --title "${prTitle}" \
```

Comment:
Can it be useful to show the resolved conflicts from `steps.merge.outputs.result` in the PR body?